### PR TITLE
Add articles documentation

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Add articles documentation
- Add article files

### CODE REVIEW

**Review by ErikSchierboom — COMMENTED** [review]
@ee7 please review https://github.com/exercism/docs/pull/406/files#diff-e29833d77834a0a9e908bbc25236fb2866ac98b158deee9f0d24a57ab1292ee2

File: building/configlet/lint.md [hunk 1]
```diff
@@ -322,6 +322,102 @@ The `config.json` file should have the following checks:
...
+- The JSON root must be an object
+- The `"articles"` key is optional, unless there is a sibling directory present (which contains the article' files)
+- The `"articles"` value must be an array of objects
+- The `"articles[].uuid"` key is required
+- The `"articles[].uuid"` value must be a unique version 4 UUID string⁶
+- The `"articles[].uuid"` value for each concept must never change
+- The `"articles[].slug"` key is required
+- The `"articles[].slug"` value must be a kebab-case string² with length <= 255
+- The `"articles[].slug"` value must have a corresponding non-empty `<slug>/content.md` file
+- The `"articles[].slug"` value must have a corresponding non-empty `<slug>/snippet.txt` file
```
Comment (ee7) [inline building/configlet/lint.md]:
```suggestion
- The `"articles[].slug"` value must have a corresponding non-empty `<slug>/snippet.md` file
```

File: building/configlet/lint.md [hunk 2]
```diff
@@ -322,6 +322,102 @@ The `config.json` file should have the following checks:
 - The `"test_runner"` key is optional
 - The `"test_runner"` value must be a boolean
 
+### Rule: exercises/practice/&lt;slug&gt;/.approaches/config.json is valid
```
Comment (ee7) [inline building/configlet/lint.md]:
```suggestion
### Rule: exercises/{concept|practice}/&lt;slug&gt;/.approaches/config.json is valid
```

Maybe something like this for all the applicable headers, if the rules for concept and practice exercises will always be the same?

**Review by ee7 — COMMENTED** [review]
`lint.md` currently refers only to `exercises/practice/{.approaches,.articles}`. Do you intend to add duplicate rules for `exercises/concept/{.approaches,.articles}`?

That is, do we want to support both approaches and articles for both kinds of exercise? No track has an approach/article for a concept exercise currently, but the [docs for concept exercises](https://github.com/exercism/docs/blob/f18232674b62025aec37e2c55cb1281e7e6618a2/building/tracks/concept-exercises.md#file-approachesintroductionmd) refer to approaches.

(I haven't reviewed properly yet)

**Review by iHiD — APPROVED** [review]

**Pull request merged**